extensions/wikia/TrackingOptIn/js/trackingOptIn.js

```diff
@@ -19,7 +19,15 @@ define('wikia.trackingOptIn', [
 	function init() {
 		if (instantGlobals.wgEnableTrackingOptInModal) {
 			log('Using tracking opt in modal', log.levels.info, logGroup);
-			var instance = trackingOptInModal.init({
+
+			var usapiEnabled =
+				window &&
```

Comment:
What a beautiful use for chaining operator and nullish coalescing it would be. `(window?.location?.search ?? []).indexOf('icUSPrivacyApi=1') !== -1 `

Reply:
BTW why do we need to check if `window.location.search` exists?

Reply:
I agree that for Node.js environment that would make sense. Is this piece of code running in Node.js?

Reply:
Actually you caught me on code copy-pasting. In ad-engine this part of code is used in Node environment, here it looks that it's not. We can remove it, but this or next sprint we'll remove this check completely so I'll leave it as safe-check since tracking-opt-in mechanism is super important.